src/radio/radio-controller.ts

```diff
@@ -165,4 +166,101 @@ export class RadioController {
   }
+
+  /**
+   * Sends an HTTP GET request using the provided parameters.
+   * @param host  The address to send the HTTP GET request to.
+   * @param port  The port to send the HTTP GET request to.
+   * @param path  The path to send the HTTP GET request to.
+   * @param timeout  How long to wait before rejecting with a timeout.
+   */
+  public async sendHttpGet(
+    host: string,
+    port: number,
+    path: string,
+    timeout = RADIO_TIMEOUT
+  ): Promise<smarthome.DataFlow.HttpResponse> {
+    const options = {
+      host,
+      port,
+      path,
+    };
+    const httpResponse = new Promise<smarthome.DataFlow.HttpResponse>(
+      resolve => {
+        http.get(options, res => {
+          const {statusCode} = res;
+          if (statusCode !== 200) {
+            throw new Error(`Request Failed.\nStatus Code: ${statusCode}`);
+          }
+          res.setEncoding('utf8');
+          res.on('data', chunk => {
+            const httpResponse = new HttpResponse(chunk, statusCode);
+            resolve(httpResponse);
+          });
+        });
+      }
+    );
+    // Timeout if still waiting for a response.
+    return Promise.race([
+      httpResponse,
+      this.createTimeoutPromise(timeout).then(() => {
+        throw new Error(
+          'UDP send timed out after ' + RADIO_TIMEOUT.toString() + 'ms.'
+        );
+      }),
+    ]);
+  }
+
+  /**
+   * Sends an HTTP POST request using the provided parameters.
+   * @param host  The address to send the HTTP POST request to.
+   * @param port  The port to send the HTTP POST request to.
+   * @param path  The path to send the HTTP POST request to.
+   * @param timeout  How long to wait before rejecting with a timeout.
+   */
+  public async sendHttpPost(
+    host: string,
+    port: number,
+    path: string,
+    dataType: string,
+    data: string,
+    timeout = RADIO_TIMEOUT
+  ): Promise<smarthome.DataFlow.HttpResponse> {
+    const options = {
+      host,
+      port,
+      path,
+      method: 'POST',
+      headers: {
+        'Content-Type': dataType,
+        'Content-Length': Buffer.byteLength(data),
+      },
+    };
+    const httpResponse = new Promise<smarthome.DataFlow.HttpResponse>(
+      resolve => {
+        const postRequest = http.request(options, res => {
+          const {statusCode} = res;
+          if (statusCode !== 200) {
+            throw new Error(`Request Failed.\nStatus Code: ${statusCode}`);
+          }
+          res.setEncoding('utf8');
+          res.on('data', chunk => {
+            const httpResponse = new HttpResponse(chunk, statusCode);
+            resolve(httpResponse);
+          });
+        });
+        postRequest.write(data);
+        postRequest.end();
+      }
+    );
+    // Timeout if still waiting for a response.
+    return Promise.race([
+      httpResponse,
+      this.createTimeoutPromise(timeout).then(() => {
+        throw new Error(
+          'UDP send timed out after ' + RADIO_TIMEOUT.toString() + 'ms.'
```

Comment:
Same note here on the error string.

Reply:
Used `timeout` and string templating.  Also fixed error message to `HTTP POST request timed out...`